src/vm/mod.rs

```diff
@@ -753,4 +753,20 @@ impl<T> Forth<T> {
         entry.finish(name, Self::variable);
         Ok(0)
     }
-}
\ No newline at end of file
+}
+
+/// # Safety
```

Comment:
Take it or leave it, you might want to move this (or reference this) in the safety invariants for `new` and `new_async`. Otherwise looks reasonable to me!

Reply:
i was kinda thinking i'd add real docs there Later. but, that might have just been because i didn't actually want to do it